src/scanspec/core.py

```diff
@@ -516,6 +517,30 @@ def squash_frames(stack: List[Frames[Axis]], check_path_changes=True) -> Frames[
     return squashed
 
 
+class DimensionInfo(Generic[Axis]):
+    def __init__(
+        self,
+        axes: Tuple[Tuple[Axis, ...]],
+        shape: Tuple[int, ...],
+        snaked: Tuple[bool, ...] = None,
+    ):
+        self._axes = axes
+        self._shape = shape
+        self._snaked = snaked or (False,) * len(shape)
+
+    @property
+    def axes(self) -> Tuple[Tuple[Axis, ...]]:
+        return self._axes
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return self._shape
+
+    @property
+    def snaked(self) -> Tuple[bool, ...]:
+        return self._snaked
+
+
```

Comment:
It's not in keeping with the other classes in this file, which is why I didn't make a dataclass. it may be a holdover from worse dataclass Generic handling? I'll see how it behaves and maybe make a PR to move Frame etc. over to be dataclasses: the specs are dataclasses but the Frames etc. are not.

RE: Dimensions vs DimensionInfo: Dimension has meaning from SPG that is already present in some of the docs. This object is not a Dimension or collection of Dimensions, it describes some Dimensions. 
Each component of axes/shape/snaked refers to the behaviour of a single Dimension. Dimension**s**Info for maybe

Reply:
Could also suggest `Dimensionality`

Reply:
Dimensionality I like

Reply:
Good point, none of the other classes in core are dataclasses, but they do all contain a lot of logic, so I think that's correct. This almost purely holds data.
From discussion with Tom, he's leaning towards suggesting that there be no default for `snaked` and  that be dealt with downstream. That would then make this a pure data class.